lib/evmone/execution_state.hpp

```diff
@@ -24,32 +24,22 @@ using bytes = std::basic_string<uint8_t>;
 using bytes_view = std::basic_string_view<uint8_t>;
 
 
-/// The stack for 256-bit EVM words.
-///
-/// This implementation reserves memory inplace for all possible stack items (1024),
-/// so this type is big. Make sure it is allocated on heap.
-struct Stack
+/// The view/controller for EVM stack.
+class Stack
 {
-    /// The maximum number of stack items.
-    static constexpr auto limit = 1024;
-
+public:
     /// The pointer to the top item.
     /// This is never nullptr and when stack is empty it points to bottom(),
```

Comment:
It's not quite true now, as it's nullptr until `reset()` is called for the first time... Maybe comment somewhere that this class assumes that `reset()` is called after constructor before any other method?